### PR TITLE
Update EVMC and evmone dependencies to cover Cancun

### LINKED ISSUE #395: evmone: update evmone to get support for latest revisions
For completeness, the evmzero dependency should within Tosca should be updated to reference a version supporting the latest EVM bytecode revisions.

### LINKED ISSUE #419: evmc: update to version supporting new releases
The current version of EMVC used by the evmzero and evmone implementations to bridge between the Go and C++ code base does not provide interfaces sufficient for supporting newer revisions. In particular, no TSTORE and TLOAD operations are supported.

To enable the development and testing of newer revisions, the EVMC dependency needs to be updated. This includes the following steps:
- [ ] pick a version of the EVMC library to target
- [ ] port Tosca's modifications required to integrate EVMC VMs into the CT tool to the targeted version
- [ ] update the dependency in Tosca
- [ ] update the `evmzero` and `evmone` implementation to be compatible with this new EVMC library
- [ ] move the modified EVMC repository from a personal repository to a organisation owned repository

### PR DESCRIPTION
**Pull request by HerbertJordan**
This PR updates the following external dependencies:
- `evmc` is updated to version 11.0.1 and the repository reference is updated to Fantom internal source; the needed modifications for Tosca got re-worked to be a backward-compatible extensions (see [delta](https://github.com/Fantom-foundation/evmc/compare/master...Fantom-foundation:evmc:tosca-extensions?expand=1)); in particular, the code hash is now passed through a back-ward compatible mechanism, enabling the integration of an unmodified `evmone`
- `evmone` was updated to the latest version to be compatible to Tosca's evmc version

As part of the restructuring, interfaces for forwarding Transient storage operations have been added.

Fixes #395 
Fixes #419

### CODE REVIEW

File: go/vm/interpreter_mock.go [hunk 1]
```diff
@@ -1,22 +1,5 @@
...
-//
-// Use of this software is governed by the Business Source License included
-// in the LICENSE file and at fantom.foundation/bsl11.
-//
-// Change Date: 2028-4-16
-//
-// On the date above, in accordance with the Business Source License, use
-// of this software will be governed by the GNU Lesser General Public Licence v3
-//
-
```
Comment (LuisPH3) [inline go/vm/interpreter_mock.go:12]: I guess it is not a big deal if we don't license generated code.
Is it there a make command to run the code generator?

Comment (HerbertJordan) [reply go/vm/interpreter_mock.go]: There is no make command, but a go command: `go generate ./...`

About the license ... we have not ported the license header script from Carmen yet (they have developed one we can hopefully reuse). I will fix the header manually for now, but we should port it at some point.

I also used a newer version of `gomock` to match the version used for generating the previous files. This reduces noise in the generated files.

**Review by facuMH — APPROVED** [review]
thanks for the upgrade.

**Review by LuisPH3 — DISMISSED** [review]

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -22,9 +22,10 @@ jobs:
       with:
         submodules: recursive
 
-    - uses: egor-tensin/setup-clang@v1
+    - name: Set up GCC
+      uses: egor-tensin/setup-gcc@v1
       with:
-        version: 14
+        version: 11
```
[the remark this answers is not in the document]
Comment (HerbertJordan) [reply .github/workflows/go.yml:28]: I am happy to have found a configuration that works. We can experiment with updated versions independently.

**Review by facuMH — APPROVED** [review]

File: .github/workflows/go.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (HerbertJordan) [reply .github/workflows/go.yml]: I updated it now, since it seems to work and I needed another change to re-trigger the CI tests.

**Review by simonlechner — APPROVED** [review]
Nice update.

**Review by LuisPH3 — APPROVED** [review]
:+1:

**Pull request merged**